synapse/handlers/room.py

```diff
@@ -1787,7 +1787,7 @@ def __init__(self, hs: "HomeServer"):
     async def shutdown_room(
         self,
         room_id: str,
-        requester_user_id: str,
+        requester_user_id: Optional[str],
```

Comment:
Did you test that users in a room would properly be kicked with `None` here ?

I think we should be fine because if `None` it is probably just visible as a normal leave from the user itself, but I am unsure about it.

Reply:
in my testing the room is deleted so quickly I can't see if it generated any leaves — though on the other hand, through the module API I'm told about new events so they must be getting generated.

However yeah, it's the user that leaves rather than the admin user kicking them — the admin user doesn't have to even be in the room.